platform/android/MapboxGLAndroidSDK/src/main/java/com/mapbox/mapboxsdk/geometry/LatLngBounds.java

```diff
@@ -337,10 +332,6 @@ private static void checkParams(
       || latSouth > GeometryConstants.MAX_LATITUDE || latSouth < GeometryConstants.MIN_LATITUDE) {
       throw new IllegalArgumentException("latitude must be between -90 and 90");
     }
-
-    if (latNorth < latSouth) {
-      throw new IllegalArgumentException("latNorth cannot be less than latSouth");
-    }
```

Comment:
Why do we need to remove this check?
I think we should add a similar check for lonEast being always >= lonWest

Reply:
Also LatLngBounds union, intersect, getCenter, getLongitudeSpan, fromLatLngs can be simplified a lot as  lonEast >= lonWest

Reply:
GeometryConstants for longitude need to be changed. Do we allow longitude to infinity ?

Reply:
It seems like what you mean @osana has already been done in https://github.com/mapbox/mapbox-gl-native/pull/13419. Could you finish up that PR and possibly port @tobrun's example while removing the `wrap` object? This would give us the desired effect, is that right?

Reply:
Another place that should be adjusted (I would think) is the Projection.getVisibleRegion() as it can go over dateline and be rotated (on top of that)

Reply:
@LukasPaczos  I absolutely agree that #13419 and this one compliment one another.  I can move over @tobrun's example or have both PRs merged - either way is fine  by me

Reply:
`Why do we need to remove this check?` 
my bad, this was re-added. 

`I think we should add a similar check for lonEast being always >= lonWest`
sure, added in follow up commit

`Also LatLngBounds union, intersect, getCenter, getLongitudeSpan, fromLatLngs can be simplified a lot as lonEast >= lonWest`

Simplified, so not a requirement for this PR?

`GeometryConstants for longitude need to be changed. Do we allow longitude to infinity ?`

 I went with Double.MAX_VALUE in a follow up commit, not seeing that infinity will unlock any use-cases.